system-test/spanner.js

```diff
@@ -2198,6 +2201,305 @@ describe('Spanner', function() {
+
+              var row = rows[0];
+              assert.strictEqual(row[0].value, true);
+
+              done();
+            });
+          });
+
+          it('should allow nullness checks', function(done) {
+            var query = {
+              sql: 'SELECT @structParam IS NULL',
+              params: {
+                structParam: Spanner.struct({
+                  userf: 'bob',
+                  threadf: Spanner.int(1),
+                }),
+              },
+            };
+
+            database.run(query, function(err, rows) {
+              assert.ifError(err);
+
+              var row = rows[0];
+              assert.strictEqual(row[0].value, false);
+
+              done();
+            });
+          });
+
+          it('should allow an array of structs with null fields', function(done) {
+            var query = {
+              sql: 'SELECT a.threadid FROM UNNEST(@structParam.arraysf) a',
+              params: {
+                structParam: Spanner.struct({
+                  intf: Spanner.int(10),
+                  arraysf: null,
+                }),
+              },
+              types: {
+                structParam: {
+                  type: 'struct',
+                  fields: [
+                    {
+                      name: 'intf',
+                      type: 'int64',
+                    },
+                    {
+                      name: 'arraysf',
+                      type: 'array',
+                      child: {
+                        type: 'struct',
+                        fields: [
+                          {
+                            name: 'threadid',
+                            type: 'int64',
+                          },
+                        ],
+                      },
+                    },
+                  ],
+                },
+              },
+            };
+
+            database.run(query, function(err, rows) {
+              assert.ifError(err);
+              assert.strictEqual(rows.length, 0);
+
+              done();
+            });
+          });
+
+          it('should allow a null array of structs', function(done) {
+            var query = {
+              sql: 'SELECT a.threadid FROM UNNEST(@structParamArray) a',
+              params: {
+                structParamArray: null,
+              },
+              types: {
+                structParamArray: {
+                  type: 'array',
+                  child: {
+                    type: 'struct',
+                    fields: [
+                      {
+                        name: 'threadid',
+                        type: 'int64',
+                      },
+                    ],
+                  },
+                },
+              },
+            };
+
+            database.run(query, function(err, rows) {
+              assert.ifError(err);
+              assert.strictEqual(rows.length, 0);
+              done();
+            });
+          });
+        });
       });
```

Comment:
what about structs with duplicate field names and missing field names?

Reply:
Struct objects in the Node client are always Arrays containing objects with a name field, if the struct with duplicate/empty names comes back from the API we handle it for them. We can also add a `Struct.fromArray` method for users who want to build objects like that themselves.

Reply:
It would be nice to add integration tests for these two scenarios.

Reply:
@adirastogi I went ahead and added the requested tests in [this commit](https://github.com/googleapis/nodejs-spanner/pull/178/commits/a988aa00cc8765c6d6e69f3da7804d848d3d1a65). PTAL!